monkey/monkey_island/cc/services/remote_run_aws.py

```diff
@@ -23,15 +25,8 @@ def init():
         :return: None
         """
         if RemoteRunAwsService.aws_instance is None:
-            RemoteRunAwsService.try_init_aws_instance()
-
-    @staticmethod
-    def try_init_aws_instance():
-        # noinspection PyBroadException
-        try:
-            RemoteRunAwsService.aws_instance = AwsInstance()
-        except Exception:
-            logger.error("Failed init aws instance. Exception info: ", exc_info=True)
+            with aws_lock:
+                RemoteRunAwsService.aws_instance = AwsInstance()
```

Comment:
I think this might be better using an event than a lock:

```python
from threading import Event
...
aws_initialization_complete = Event()
...
        if RemoteRunAwsService.aws_instance is None:
            RemoteRunAwsService.aws_instance = AwsInstance()
            aws_initialization_complete.set()
...
    def is_running_on_aws():
        aws_initialization_complete.wait(CHOOSE_APPROPRIATE_TIMEOUT)
        return RemoteRunAwsService.aws_instance.is_instance()
```